src/USD.NET/serialization/Scene.cs

```diff
@@ -745,10 +745,12 @@ public enum UpAxes {
     private pxr.UsdPrim GetUsdPrim(SdfPath path) {
       UsdPrim prim;
       lock(m_stageLock) {
-      if (!m_primMap.TryGetValue(path, out prim) || !prim.IsValid()) {
-        prim = Stage.GetPrimAtPath(path);
-        m_primMap[path] = prim;
-      }
+        if (!m_primMap.TryGetValue(path, out prim) || !prim.IsValid()) {
+          prim = Stage.GetPrimAtPath(path);
+          if (prim.IsValid()) {
+            m_primMap[path] = prim;
+          }
```

Comment:
If the prim isn't valid, should the path be cleared here to keep the primMap accurate?

Reply:
That's a good question... I don't know ^^. What would be the case where it happens?... If a prim is removed on disk maybe?? Then, when refreshing... the primMap wouldn't be accurate?

Reply:
Yeah, that's what I was thinking -- like a file is regenerated and some prims are removed, or some flavor of that.

I don't think this is a /bug/ but I think it would be a good axiom to enforce that the prim map is always either pointing to a valid prim or has no entry.

Reply:
(It should be super easy to test this with an ascii file, just delete the prim and see what happens on reload at that point)

Reply:
@jcowles What is the benefit of the primMap here? Stage.GetPrimAtPath is a sdf operation and will be cached anyway so it should be fast. 
If we keep a local cache in Scene I can't see a reliable way to ensure the cache stays in sync while we expose the stage as public. We could provide our own Scene.RemovePrim and Scene.Reload but nothing prevents the users to call scene.Stage.RemovePrim or scene.Stage.Reload ...

Reply:
I believe the original motivation was to avoid the C++/C# roundtrip just to get a prim, though it seems silly now.

I wonder if we can just kill it entirely -- given that this requires grabbing a lock, seems like that would be nice.

Reply:
Created a ticket for this one: https://jira.unity3d.com/browse/USDU-135